test/units/modules/network/eos/test_eos_acls.py

```diff
@@ -0,0 +1,276 @@
+            filename = 'eos_acls_config.cfg'
+
+        def load_from_file(*args, **kwargs):
+            output = load_fixture(filename)
+            return output
+
+        self.execute_show_command.side_effect = load_from_file
+
+    def test_eos_acls_merged(self):
+        set_module_args(
+            dict(config=[
+                dict(afi="ipv6",
+                     acls=[
+                         dict(name="test2",
+                              standard="true",
+                              aces=[
+                                  dict(sequence="10",
+                                       grant="permit",
+                                       protocol="ospf",
+                                       source=dict(subnet_address="30.2.0.0/8"),
+                                       destination=dict(any="true"),
+                                       log="true")
+                              ])
+                     ])
+            ], state="merged"))
+        commands = ['ipv6 access-list standard test2', '10 permit ospf 30.2.0.0/8 any log']
+        result = self.execute_module(changed=True, commands=commands)
+
+    def test_eos_acls_merged_idempotent(self):
+        set_module_args(
+            dict(config=[
+                dict(afi="ipv4",
+                     acls=[
+                         dict(name="test1",
+                              aces=[
+                                  dict(sequence="35",
+                                       grant="deny",
+                                       protocol="tcp",
+                                       source=dict(subnet_address="20.0.0.0/8"),
+                                       destination=dict(any="true"),
+                                       log="true"),
+                                  dict(grant="permit",
+                                       source=dict(any="true"),
+                                       destination=dict(any="true"),
+                                       protocol=6)
+                              ])
+                     ])
+            ], state="merged"))
+        result = self.execute_module(changed=False, commands=[])
+
+    def test_eos_acls_replaced(self):
+        set_module_args(
+            dict(config=[
+                dict(afi="ipv4",
+                     acls=[
+                         dict(name="test1",
+                              aces=[
+                                  dict(sequence="10",
+                                       grant="permit",
+                                       protocol="ospf",
+                                       source=dict(subnet_address="30.2.0.0/8"),
+                                       destination=dict(any="true"),
+                                       log="true")
+                              ])
+                     ])
+            ], state="replaced"))
+        commands = ['no ip access-list test1', 'ip access-list test1', '10 permit ospf 30.2.0.0/8 any log']
```

Comment:
Isn't the 'no ip access-list test1' redundant in this case?

Reply:
When there are many aces under an access-list, the replace operation , will negate the access-list as a whole and the reconfigure it, with the given ace(s). This seemed to be a simpler and straight forward approach.